appveyor.yml

```diff
@@ -19,20 +25,36 @@ for:
 # "fall back" configuration for all other branches
 # no "branches" section defined
 configuration: Debug
+install:
+  - choco install opencover.portable
+  - dotnet tool install --global dotnet-sonarscanner
 before_build:
   - cmd: nuget restore
-build:
-  project: CDP4-SDK.sln
-test:
-  assemblies:
-    only:
-      - '**\bin\$(configuration)\net472\*.Tests.dll'
-      - '**\bin\$(configuration)\netstandard1.6\*.Tests.dll'
-      - '**\bin\$(configuration)\netstandard2.0\*.Tests.dll'
-  categories:
-    except:
-      - WebServicesDependent
-      - AppVeyorExclusion
+  - ps: >-
+      if ($env:APPVEYOR_PULL_REQUEST_NUMBER){
+        Write-Host Scanning PR: $env:APPVEYOR_PULL_REQUEST_NUMBER
+        dotnet sonarscanner begin /key:$env:SONARCLOUD_PROJECT_KEY /o:$env:SONARCLOUD_ORGANIZATION /d:sonar.host.url="https://sonarcloud.io" /d:sonar.login=$env:SONARCLOUD_TOKEN /d:sonar.pullrequest.key=$env:APPVEYOR_PULL_REQUEST_NUMBER /d:sonar.pullrequest.provider=GitHub /d:sonar.pullrequest.github.repository=$env:APPVEYOR_REPO_NAME
+      }
+      else {
+        Write-Host Scanning branch: $env:APPVEYOR_REPO_BRANCH
+        dotnet sonarscanner begin /key:$env:SONARCLOUD_PROJECT_KEY /o:$env:SONARCLOUD_ORGANIZATION /d:sonar.host.url="https://sonarcloud.io" /d:sonar.login=$env:SONARCLOUD_TOKEN
+      }
+build_script:
+  - cmd: dotnet build CDP4-SDK.sln --configuration Debug -v q
+test_script:
+    - >
+      OpenCover.Console.exe
+      -log:Error
+      -register
+      -target:"dotnet"
+      -targetargs:"test --framework net45 --filter=""(TestCategory!~WebServicesDependent) & (TestCategory!~AppVeyorExclusion)"" --configuration %CONFIGURATION% -v q"
```

Comment:
can you include one of the netstandard ones here as well. There is a codepath difference between the .NET Framework and .NET Standard